pkg/auth/authorizer/abac/abac.go

```diff
@@ -52,7 +53,46 @@ func (p policyLoadError) Error() string {
 type policyList []*api.Policy
 
 // TODO: Have policies be created via an API call and stored in REST storage.
-func NewFromFile(path string) (policyList, error) {
+func NewFromFile(path string) (*policyList, error) {
+
+	pl, err := newFromFile(path, nil)
+	if err != nil {
+		return nil, err
+	}
+
+	// Create an empty watcher
+	watcher, err := fsnotify.NewWatcher()
+	if err != nil {
+		return nil, err
+	}
+
+	// Watch the policy file
+	err = watcher.Add(path)
+	if err != nil {
+		return nil, err
+	}
+
+	// On write operations on the policy file, reload policy file
+	go func() {
+		for {
+			select {
+			case ev := <-watcher.Events:
+				switch ev.Op {
+				case fsnotify.Write:
```

Comment:
If you make a copy, edit it, and then [rename](http://man7.org/linux/man-pages/man2/rename.2.html) using the rename syscall (mv should work too, I think), then the new file will appear atomically.  

You would need to document to users this requirement.

If you are polling, then this works just the way you want it to: the apiserver never sees a partial file.  

I don't see fsnotify helping you much here, versus polling every few seconds (assuming policy files are reasonably sized).

You still need to handle the case of an incorrect file.  Should you crash, as is currently the case when an incorrect file is provided?  Or keep using the old one?  How to surface this error to the user?